File: app/src/main/java/com/wang17/myphone/util/_DialogUtils.java
package com.wang17.myphone.util;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.CalendarView;
import android.widget.EditText;
import android.widget.Switch;

import com.wang17.myphone.R;
import com.wang17.myphone.callback.MyCallback;
import com.wang17.myphone.model.database.BankToDo;
import com.wang17.myphone.model.DateTime;
import com.wang17.myphone.model.database.Setting;
import com.wang17.myphone.widget.MyWidgetProvider;

import java.lang.reflect.Field;
import java.text.DecimalFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

import static com.wang17.myphone.util._Utils.e;

public class _DialogUtils {

    private static final String TAG = "wangsc";

    /**
     * 新建todo项
     */
    public static void addTodoDialog(final Context context, final MyCallback callback) {
        final View view = LayoutInflater.from(context).inflate(R.layout.dialog_add_todo, null);
        CalendarView calendarViewDate = view.findViewById(R.id.calendarView_date);
        EditText editTextTitle = view.findViewById(R.id.editText_title);
        EditText editTextSummery = view.findViewById(R.id.editText_summery);
        EditText editTextNumber = view.findViewById(R.id.editText_number);

        DateTime dateTime = new DateTime();
        calendarViewDate.setOnDateChangeListener(new CalendarView.OnDateChangeListener() {
            @Override
            public void onSelectedDayChange(@NonNull CalendarView view, int year, int month, int dayOfMonth) {
                dateTime.set(year,month,dayOfMonth);
            }
        });


        new AlertDialog.Builder(context)
                .setView(view).setCancelable(false).setPositiveButton("提交", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {

                try {

                    String bankName = editTextTitle.getText().toString();
                    double money = Double.parseDouble(editTextSummery.getText().toString());
                    String number = editTextNumber.getText().toString();

                    DataContext dataContext = new DataContext(context);

                    BankToDo bankToDo = new BankToDo(dateTime, bankName, number, money);
                    dataContext.addBankToDo(bankToDo);


                    // 发送更新广播
                    Intent intent = new Intent(MyWidgetProvider.ACTION_UPDATE_LISTVIEW);
                    context.sendBroadcast(intent);

                    if (callback != null)
                        callback.callBack();

                } catch (Exception e) {
                    _Utils.printException(context, e);
                }
            }
        }).setNegativeButton("取消", null).create().show();

    }

    /**
     * 编辑todo项
     */
    public static void editTodoDialog(final Context context, final BankToDo bankToDo, final MyCallback callback) {
        final View view = LayoutInflater.from(context).inflate(R.layout.dialog_add_todo, null);


        DataContext dataContext = new DataContext(context);
        DateTime dateTime = bankToDo.getDateTime();
        CalendarView calendarViewDate = view.findViewById(R.id.calendarView_date);
        EditText editTextTitle = view.findViewById(R.id.editText_title);
        EditText editTextSummery = view.findViewById(R.id.editText_summery);
        EditText editTextNumber = view.findViewById(R.id.editText_number);

        e("xxxxxxxxxxxxxxxxxxxxxxxx : "+bankToDo.getDateTime().toLongDateTimeString());
        calendarViewDate.setDate(bankToDo.getDateTime().getTimeInMillis());
        calendarViewDate.setOnDateChangeListener(new CalendarView.OnDateChangeListener() {
            @Override
            public void onSelectedDayChange(@NonNull CalendarView view, int year, int month, int dayOfMonth) {
                dateTime.set(year,month,dayOfMonth);
            }
        });
        editTextTitle.setText(bankToDo.getBankName());
        editTextTitle.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    editTextTitle.selectAll();
                }
            }
        });
        String format= "#,##0.00";
        if(bankToDo.getMoney()%1==0){
            format= "#,##0";
        }
        editTextSummery.setText(new DecimalFormat(format).format(bankToDo.getMoney()));
        editTextSummery.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    editTextSummery.selectAll();
                }
            }
        });
        editTextNumber.setText(bankToDo.getCardNumber());
        editTextNumber.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    editTextNumber.selectAll();
                }
            }
        });
        new AlertDialog.Builder(context).setView(view).setCancelable(false).setPositiveButton("提交", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                try {
                    String bankName = editTextTitle.getText().toString();
                    double money = Double.parseDouble(editTextSummery.getText().toString().replace(",",""));
                    String number = editTextNumber.getText().toString();

                    bankToDo.setBankName(bankName);
                    bankToDo.setDateTime(dateTime);
                    bankToDo.setMoney(money);
                    bankToDo.setCardNumber(number);
                    dataContext.editBankToDo(bankToDo);

                    // 发送更新广播
                    Intent intent = new Intent(MyWidgetProvider.ACTION_UPDATE_LISTVIEW);
                    context.sendBroadcast(intent);

                    if (callback != null)
                        callback.callBack();
                } catch (Exception e) {
                    _Utils.printException(context, e);
                }
            }
        }).setNegativeButton("取消", null).create().show();

    }
}

File: app/src/main/java/com/wang17/myphone/util/_DialogUtils.kt
package com.wang17.myphone.util

import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.support.v7.app.AlertDialog
import android.view.LayoutInflater
import android.view.View.OnFocusChangeListener
import android.widget.CalendarView
import android.widget.EditText
import com.wang17.myphone.R
import com.wang17.myphone.activity.DesktopDialogActivity
import com.wang17.myphone.callback.MyCallback
import com.wang17.myphone.database.BankToDo
import com.wang17.myphone.database.DataContext
import com.wang17.myphone.model.DateTime
import com.wang17.myphone.util._Utils.e
import com.wang17.myphone.util._Utils.printException
import com.wang17.myphone.widget.MyWidgetProvider
import java.text.DecimalFormat

object _DialogUtils {

    fun showDesktopDialog(context: Context, msg: String){
        val intent = Intent(context, DesktopDialogActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_NO_USER_ACTION
        intent.putExtra("msg", msg)
        context.startActivity(intent)
    }

    @JvmStatic
    fun showMessageBox(context: Context, msg: String){
        AlertDialog.Builder(context).setMessage(msg).show()
    }
    @JvmStatic
    fun showMessageBox(context: Context, msg: String, btnText: String){
        AlertDialog.Builder(context).setMessage(msg).setCancelable(false).setPositiveButton(btnText, null).show()
    }
    @JvmStatic
    fun showMessageBox(context: Context, msg: String, btnText: String, btnListener: DialogInterface.OnClickListener?){
        AlertDialog.Builder(context).setMessage(msg).setCancelable(false).setPositiveButton(btnText, btnListener).show()
    }
    /**
     * 新建todo项
     */
    @JvmStatic
    fun addTodoDialog(context: Context, callback: MyCallback?) {
        val view = LayoutInflater.from(context).inflate(R.layout.dialog_add_todo, null)
        val cvDate = view.findViewById<CalendarView>(R.id.cv_date)
        val etBankName = view.findViewById<EditText>(R.id.et_bankName)
        val etMoney = view.findViewById<EditText>(R.id.et_money)
        val etSummary = view.findViewById<EditText>(R.id.et_summary)
        val dateTime = DateTime()
        cvDate.setOnDateChangeListener { view, year, month, dayOfMonth -> dateTime[year, month] = dayOfMonth }
        AlertDialog.Builder(context)
                .setView(view).setCancelable(false).setPositiveButton("提交") { dialog, which ->
                    try {
                        val bankName = etBankName.text.toString()
                        val money = etMoney.text.toString().toDouble()
                        val number = etSummary.text.toString()
                        val dataContext = DataContext(context)
                        val bankToDo = BankToDo(dateTime, bankName, number, money)
                        dataContext.addBankToDo(bankToDo)


                        // 发送更新广播
                        val intent = Intent(MyWidgetProvider.ACTION_UPDATE_LISTVIEW)
                        context.sendBroadcast(intent)
                        callback?.execute()
                    } catch (e: Exception) {
                        printException(context, e)
                    }
                }.setNegativeButton("取消", null).create().show()
    }

    /**
     * 编辑todo项
     */
    @JvmStatic
    fun editTodoDialog(context: Context, bankToDo: BankToDo, callback: MyCallback?) {
        val view = LayoutInflater.from(context).inflate(R.layout.dialog_add_todo, null)
        val dataContext = DataContext(context)
        val dateTime = bankToDo.dateTime
        val calendarViewDate = view.findViewById<CalendarView>(R.id.cv_date)
        val editTextTitle = view.findViewById<EditText>(R.id.et_bankName)
        val editTextSummery = view.findViewById<EditText>(R.id.et_money)
        val editTextNumber = view.findViewById<EditText>(R.id.et_summary)
        e("xxxxxxxxxxxxxxxxxxxxxxxx : " + bankToDo.dateTime.toLongDateTimeString())
        calendarViewDate.date = bankToDo.dateTime.timeInMillis
        calendarViewDate.setOnDateChangeListener { view, year, month, dayOfMonth -> dateTime[year, month] = dayOfMonth }
        editTextTitle.setText(bankToDo.bankName)
        editTextTitle.onFocusChangeListener = OnFocusChangeListener { v, hasFocus ->
            if (hasFocus) {
                editTextTitle.selectAll()
            }
        }
        var format = "#,##0.00"
        if (bankToDo.money % 1 == 0.0) {
            format = "#,##0"
        }
        editTextSummery.setText(DecimalFormat(format).format(bankToDo.money))
        editTextSummery.onFocusChangeListener = OnFocusChangeListener { v, hasFocus ->
            if (hasFocus) {
                editTextSummery.selectAll()
            }
        }
        editTextNumber.setText(bankToDo.summary)
        editTextNumber.onFocusChangeListener = OnFocusChangeListener { v, hasFocus ->
            if (hasFocus) {
                editTextNumber.selectAll()
            }
        }
        AlertDialog.Builder(context).setView(view).setCancelable(false).setPositiveButton("提交") { dialog, which ->
            try {
                val bankName = editTextTitle.text.toString()
                val money = editTextSummery.text.toString().replace(",", "").toDouble()
                val number = editTextNumber.text.toString()
                bankToDo.bankName = bankName
                bankToDo.dateTime = dateTime
                bankToDo.money = money
                bankToDo.summary = number
                dataContext.editBankToDo(bankToDo)

                // 发送更新广播
                val intent = Intent(MyWidgetProvider.ACTION_UPDATE_LISTVIEW)
                context.sendBroadcast(intent)
                callback?.execute()
            } catch (e: Exception) {
                printException(context, e)
            }
        }.setNegativeButton("取消", null).create().show()
    }
}